src/pybamm/models/submodels/thermal/base_thermal.py

```diff
@@ -88,6 +88,12 @@ def _get_standard_coupled_variables(self, variables):
         # TODO: change full stefan-maxwell conductivity so that i_e is always
         # a Concatenation
         i_e = variables["Electrolyte current density [A.m-2]"]
+        # Special case for half cell -- i_e has to be a concatenation for this to work due to a mismatch with Q_ohm, so we make a new i_e which is a concatenation.
+        if (not isinstance(i_e, pybamm.Concatenation)) and (
+            self.options.electrode_types["negative"] == "planar"
+        ):
```

Comment:
Just for clarification, what happens outside of the `self.options.electrode_types["negative"] == "planar"` case? Are there still issues with other electrodes or electrode types? If so is there an issue for it somewhere?

Reply:
In that case, Q_rxn is a concatenation with domains that match Q_ohm, which means everything matches up and works ( I just double checked this).